### DISCUSSION TITLE
Files for using with VBA installed in the correct place?

### DISCUSSION THREAD

**Discussion by TerraD** [micjahn/ZXing.Net#387] [Q&A] [answered]
Thank you for the interop.dll and the register.cmd. Installation of ZXing is very simple [that way](https://github.com/micjahn/ZXing.Net/wiki/Using-ZXing.Net-with-VBA-(COM-Interop)) and makes using ZXing with VBA a snap! 

However I'm not sure whether installation in the Windows directory (C:\Windows) itself is state of the art. Shouldn't .dll-files be installed in C:\Windows\System32 for x86 and C:\Windows\SysWOW64 for x64?

I'm just an experienced VBA user but no expert on this matter nor do I understand the mechanisms behind regasm... For this reason I'll not fiddle with the way it is installed now - just _very_ happy it works! But maybe someone with more experience can enlighten me...

**Comment by micjahn** [comment, marked as answer]
The interop assembly and the core zxing.net assembly are built as "Any CPU" and should be runnable within an x86 and an x64 context.
You can place the assemblies at any location but it has to be in the search path. Excel, Word, etc. need a chance to find the dll, especially the zxing.dll. That was the reason why I decided to use C:\Windows.

**Comment by TerraD** [comment]
Thank you for this answer - as I stated in the beginning, I really have not big knowledge about this. However I do not like .dll installed in C:\Windows as this is hardly ever done by other programs. Your comment encouraged me to try the following:

In order to tidy up C:\Windows I followed your installation procedure as explained in [Using ZXing.Net with VBA (COM Interop)](https://github.com/micjahn/ZXing.Net/wiki/Using-ZXing.Net-with-VBA-(COM-Interop)) and moved after installation all (totally 6) zx*.* files (.dll, .pdb, .tlb, .xml) manually to C:\Windows\SysWOW64. 

This seems to work well. I'm running this on a Win10 machine and as I realized just now, it even works with Microsoft Access _32-bit_ - considering it is Access 32-bit \System32 might have been the correct folder! :-) Thus, the ZXing-files seem really tolerant regarding installation location and registering by means of regasm.exe seems tolerant regarding this too!

Thank you!
